packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -54,6 +59,64 @@ const Duration _kCursorBlinkWaitForStart = Duration(milliseconds: 150);
 // is shown in an obscured text field.
 const int _kObscureShowLatestCharCursorTicks = 3;
 
+/// Represents one "replacement" to check for, consisting of a [Pattern] to
+/// match and a generator [InlineSpanGenerator] function that creates an
+/// [InlineSpan] from a matched string.
+///
+/// The generator function is called for every match of the pattern found.
+///
+/// Typically, the generator should return a custom [TextSpan] with unique styling
+/// or a [WidgetSpan] to embed widgets within text fields.
+///
+/// {@tool snippet}
+/// In this example, all strings enclosed in {} are matched and
+/// the contents of the braces are interpreted as an image url.
+///
+/// ```dart
+/// TextEditingInlineSpanReplacement(
+///   RegExp(r'\{[\w\/\.]+\}'),
+///   (String value, TextRange range) {
+///     return WidgetSpan(
+///       child: Image.asset(value.substring(1, value.length - 1)),
+///       range: range,
+///     );
+///   },
+/// )
+/// ```
+/// {@end-tool}
+///
+/// {@tool snippet}
+/// In this simple example, the word following a # symbol is styled in blue.
+///
+/// ```dart
+/// TextEditingInlineSpanReplacement(
+///   RegExp(r'#[\w]+'),
+///   (String value, TextRange range) {
+///     return TextSpan(text: value, style: TextStyle(color: Colors.blue));
+///   },
+/// )
+/// ```
+/// {@end-tool}
```

Comment:
Can you add a note to the doc where an `TextEditingInlineSpanReplacement` would typically be used, i.e. what class takes it as a param?